README.md

```diff
@@ -1,21 +1,348 @@
-Incus
-===
+Incus [![Build Status](https://travis-ci.org/Imgur/incus.svg?branch=master)](https://travis-ci.org/Imgur/incus) [![Coverage Status](https://coveralls.io/repos/Imgur/incus/badge.svg)](https://coveralls.io/r/Imgur/incus)
+=========
 
-[![Build Status](https://travis-ci.org/Imgur/incus.svg?branch=master)](https://travis-ci.org/Imgur/incus)
-[![Coverage Status](https://coveralls.io/repos/Imgur/incus/badge.svg)](https://coveralls.io/r/Imgur/incus)
+![incus](http://i.imgur.com/7ZgRrA5.png)
 
-Incus was created by Imgur.com to be a real-time messaging server that acts as the
-intermediary between a web app and a browser.
+middleware for distributing messages via websockets, long polling, and push notifications
 
-## How it's used at Imgur
-Incus powers Imgur's real-time direct messaging, notifications (including iOS push), and
-real-time point updates on all images and comments. It's currently processing over 1,000
-events per second per machine, but can actually handle much more than that
-(limits have not yet been identifed).
+## Features
+* Websocket authentication and management
+* Long Poll fall back
+* iOS push notifications support through APNS
+* Android push notifications support through GCM
+* Routing messages to specific phone, authenticated user, or webpage url
+* Configurable option for allowing users to send messages to other users
+* Redis pub/sub and Redis List support for sending messages from an application
+* SSL support
+* Stats logging
+
+## Usage
+
+### incus.js
+Once incus is running you can include incus.js onto your site.
+Here's the basic usage of incus.js
+
+```Javascript
+var socket = new Incus('http://localhost:4000', 'UID', '/page/path');
+socket.on('connect', function() {
+    console.log('connected');
+}
+
+socket.on('Event', function(data) {
+    alert(data);
+}
+
+socket.on('Event1', function (data) {
+   console.log(data);
+}
+
+socket.on('Event2', function(data) {
+    console.log('neat');
+} 
+
+var data = {data: 'dummy-data'};
+
+$('#button').on('click', function() {
+    socket.MessageUser('Event', 'UID', data); 
+    socket.MessageAll('Event1', data);
+    socket.MessagePage('Event2', '/page/path', data);
+});
+```
+
+### Application to a web browser
+
+UID is a unique identifier for the user. It can be anywhere from an auto incremented ID to something more private such as a session id or OAuth token.
```

Comment:
suggestion: something like "If messages to users should be private from other users, then you should consider the UID a shared secret between the user and the server" - suggests something similar to the scheme we have.